### ISSUE TITLE
Installation procedure improvement

### ISSUE THREAD

**Issue by Remy-Luciani** [theodo/theodo-stack-generator#141]
With @clementhannicq , we want to make each generated project installation a great experience for the developers: it has to be short, simple because it works at first try everytime. To do so, we want to generate a Makefile executing all the necessary tasks.

Nevertheless, we have spotted some steps that need to be improve before creating such Makefile. We note them here.

Project where the generator has been used : Wero.

1. Configure provisioning in the Vagrantfile

Current situation: boot the VM, add the dev user's SSH key into the VM, execute ansible
Target: provision through Vagrant with Vagrant auth, thanks to a config like the following

```ruby
config.vm.provision "ansible" do |ansible|
  ansible.playbook = "devops/provisioning/playbook.yml"
  ansible.inventory_path = "devops/provisioning/hosts/vagrant"
  ansible.limit = 'all'
  ansible.ask_vault_pass = true
  if ENV.has_key?('ANSIBLE_TAGS')
    ansible.tags = ENV['ANSIBLE_TAGS']
  end
end
```

2. Rework file structure into the Guest machine

Current situation: symlinks from frontend apps must be linked to `api/public`
Target: apps are built directly in the webserver root directory

**Comment by benja-M-1** [comment]
@Remy-Luciani this was the case before @Miniplop changed it here: https://github.com/theodo/theodo-stack-generator/commit/a2e2b0cf59f5c2aed72ccebd872f1a06430348a0#diff-1e2a76fddfcdabb9d06d8acc3ebefa42.

@Miniplop do you remember why you removed the provisioning command from the Vagrantfile?

**Comment by benja-M-1** [comment]
@Miniplop ?

**Comment by clementhannicq** [comment]
@benja-M-1 Seen with @Miniplop one of the problem solved by separating those is to enable people to use the ansible playbook to provision a staging environment.

**Comment by benja-M-1** [comment]
@clementhannicq hum...  IMO the provisioning command should be at least documented to do it when needed, at best in a makefile. But it is not a good reason enough to me to adding this command to run when building the VM. 

The provisioning could be rerun with ```vagrant provision```. We could add in the docs that the provisioning could be rerun with the ansible command with specific tags.

Do you think you can submit a PR for that?

**Comment by clementhannicq** [comment]
@benja-M-1 I agree with you on that, but probably because I never encountered the issue. That would basically be reverting @Miniplop 's PR ? From what I understand separating those is something you both agreed on.

I'll have a look later, there's other things I want to do first on the generator.
